src/main/java/com/buglife/sdk/reporting/SubmitReportLegacyService.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (C) 2017 Buglife, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.buglife.sdk.reporting;
+
+import android.app.IntentService;
+import android.content.Context;
+import android.content.Intent;
+import android.support.annotation.Nullable;
+
+import com.android.volley.NoConnectionError;
+import com.buglife.sdk.Log;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.Closeable;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class SubmitReportLegacyService extends IntentService {
+    private static final String KEY_EXTRA_JSON_REPORT = "json_report";
+    private SubmitReportTask mTask;
+
+    public static void start(Context context, String jsonReport) {
+        Intent intent = new Intent(context, SubmitReportLegacyService.class);
+        intent.putExtra(KEY_EXTRA_JSON_REPORT, jsonReport);
+        context.startService(intent);
+    }
+
+    public static void start(Context context) {
+        Intent intent = new Intent(context, SubmitReportLegacyService.class);
+        context.startService(intent);
+    }
+
+    public SubmitReportLegacyService() {
+        super("SubmitReportLegacyService");
+    }
+
+    @Override public void onCreate() {
+        super.onCreate();
+        mTask = new SubmitReportTask(getApplicationContext());
+    }
+
+    @Override protected void onHandleIntent(@Nullable Intent intent) {
+        File cacheFile = getReportsCacheFile(getApplicationContext());
+        if (!cacheFile.exists()) {
+            Log.i("Reports cache file doesn't exist! No reports to submit.");
```

Comment:
Whenever I try to submit a report on my KitKat device, I see this get logged & my bug report is never actually submitted. It seems the `reports` directory for legacy submission is never actually created, so this method returns early. Before I went ahead and changed anything, I wanted to verify that the intent here (no pun intended) is to submit the "current" bug report, as well as all cached bug reports?

Reply:
@schukin This is a good catch. I had changed the behavior slightly so this check is no longer valid.